src/briefcase/integrations/docker.py

```diff
@@ -99,9 +151,14 @@ def verify_docker(command):
 If you have installed Docker recently and are still getting this error, you may
 need to restart your terminal session.
 """
+
+    # short circuit since already verified and available
+    if hasattr(tools, "docker"):
+        return
```

Comment:
Similar to the apple tools, this caches a tool instead of a flag. I separated this from verifying Docker for the app to avoid verifying Docker exists and can run multiple times.